src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        if (contactsToUpdate.size() > 0) {
+            update contactsToUpdate;
+        }
     }
 
     /*******************************************************************************************************
-    * @description Deletes the accounts for contacts that have been deleted if they are empty (no ops, no hh members)
-    * @param contacts the list of Contacts that need to be considered
-    * @param dmlWrapper to hold the Accounts that need updating.
-    * @return void
+    * @description Updates an Account's naming, primary contact, and rollups.
+    * @param accountIdsToUpdate the list of Account (Id's) that needs update.
+    * @param dmlWrapper to hold the Accounts that needs update.
     */
-    public void deleteContactAccountsIfEmpty(Contact[] contacts, DmlWrapper dmlWrapper) {
-        List<Id> accountIds = new List<Id>();
-        List<Account> accountsOnContacts = new List<Account>();
-        List<SObject> accountsForDeletion = new List<SObject>();
-        List<Id> listAccountIdUpdate = new list<Id>();
+    private void updateParentAcc(List<Id> accountIdsToUpdate, DmlWrapper dmlWrapper) {
+        resetPrimaryContactForAccount(accountIdsToUpdate, dmlWrapper);
+
+        //Inject Adv logic to do the rollup for Account
+        if (Advancement_Info.useAdv() && Advancement_Info.getApiHEDA() != null) {
+            Advancement_Info.getApiHEDA().rollupAccounts(accountIdsToUpdate);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description sets the Primary Account for each Account in the list of Accounts
+    * @param accountIdsToUpdate a list of Account (Id's) that needs update.
+    * @param dmlWrapper to hold the Accounts that needs update.
+    */
+    private void resetPrimaryContactForAccount(List<Id> accountIdsToUpdate, DmlWrapper dmlWrapper) {
+        Set<Id> primaryContactId = new Set<Id>();
+        Set<Id> accountsToResetIds = new Set<Id>();
+        List<Account> accountsToUpdate = new List<Account>();
+
+        //Builds a set of primary Contact Ids from the Account given existing AccountIds
+        List<Account> listAccount = [SELECT Primary_Contact__c,
+                                     (SELECT Id, Name FROM Contacts LIMIT 1)
+                                     FROM Account
+                                     WHERE Id IN :accountIdsToUpdate];
 
-        if (contacts.size() > 0) {
-            for (Contact thisContact : contacts) {
-                accountIds.add(thisContact.accountId);
+        //Retrieves the Primary_Contact__c from the Account
+        for (Account acc : listAccount) {
+            if (acc.Primary_Contact__c != NULL)
+                primaryContactId.add(acc.Primary_Contact__c);
+        }
+
+        if (primaryContactId.size() > 0) {
+            for (Account acc : [SELECT Primary_Contact__c,
+                                (SELECT Id, Name FROM Contacts WHERE Id IN :primaryContactId LIMIT 1)
```

Comment:
same question here regarding the subquery (it could be that I'm just not there yet)

Reply:
@lifewithryan , from this line https://github.com/SalesforceFoundation/EDA/blob/0042f119431dcf0fe45ffe6417034d80332e4463/src/classes/ACCT_IndividualAccounts_TDTM.cls#L671 's query, it seems a bit redundant from line 659 and 665. However, this particular soql for loop is specifically querying for a Contact that's associated to the AccountId that matches the Primary_Contact__c value from the Account record. It then does a check if there's no Contact record that's associated to the AccountId that matches the Id on the Primary_Contact__c on the Account OR if the Primary_Contact__c field is blank on the Account.